Backpack/BottomSheet/Classes/BottomSheet.swift

```diff
@@ -0,0 +1,142 @@
+    }
+    
+    private lazy var floatingPanelController: BackpackFloatingPanelController = {
+        let panel = BackpackFloatingPanelController(delegate: self)
+        panel.surfaceView.cornerRadius = 24.0
+        panel.isRemovalInteractionEnabled = true
+        
+        // We do this to hold a strong reference to `BottomSheet` and force it
+        // to exist as long as `floatingPanelController` exists.
+        // Reference will be cleaned up by `floatingPanelController` when
+        // it's dismissed, to avoid a reference cycle.
+        panel.backpackBottomSheet = self
+        
+        return panel
+    }()
+    
+    /// View controller that will be presented when calling
+    /// `present(in: _, animated: _, completion: _)`.
+    /// It can also be presented using UIKit's native presentation API.
+    public var viewControllerToPresent: UIViewController {
+        get {
+            return floatingPanelController
+        }
+    }
+    
+    /// View controller contained in the bottom sheet.
+    public var contentViewController: UIViewController? {
+        get {
+            return floatingPanelController.contentViewController
+        }
+    }
+    
+    /// Fixed bottom section. Can only be passed when initializing
+    /// the bottom sheet.
+    public var bottomSectionViewController: UIViewController? {
+        get {
+            return floatingPanelController.bottomSectionViewController
+        }
+    }
+    
+    /// This closure will be executed once the bottom sheet has been
+    /// fully dismissed.
+    public var onDismissed: (() -> Void)? {
+        get {
+            return floatingPanelController.onDismissed
+        }
+        set {
+            floatingPanelController.onDismissed = newValue
+        }
+    }
+    
+    /// Instantiates a `BottomSheet`.
+    ///
+    /// - Parameters:
+    ///   - contentViewController: Content of the bottom sheet.
+    ///   - scrollViewToTrack: The bottom sheet uses this scroll view's gesture recognizer
+    ///     instead of creating a new one. This allows the scroll view to start offseting its
+    ///     content only after the bottom sheet has reached its full screen position.
+    ///   - bottomSectionViewController: Optional. A bottom section can be provided so that it is
+    ///     always visible regardless of the bottom sheet position. Useful for providing actions
+    ///     that should be accessible at all times. A top shadow is automatically added so that
+    ///     it integrates better with the content of the bottom sheet.
+    ///     Note: Safe Area should be taken into account in the bottom section's inner constraints.
+    public init(contentViewController: UIViewController, scrollViewToTrack: UIScrollView, bottomSectionViewController: UIViewController? = nil) {
+        super.init()
+        floatingPanelController.contentViewController = contentViewController
+        floatingPanelController.track(scrollView: scrollViewToTrack)
```

Comment:
Right now, yes. If a scroll view isn't provided then we can assume that the content of the bottom sheet can fit in the screen without scrolling. So in that case, we might not want a hardcoded size (386.0) but rather want the bottom sheet to use the intrinsic size of the content.

There are also some other cases I can think of, but none are supported right now by our custom wrapper. By making the scroll view mandatory, we can postpone working on these for a later time when we actually need it.

Starting with a limited API and open it up later is no problem. But we would have a problem if we start we an open API (that makes the scroll view optional) and then we realize that it doesn't work as expected and we can't restrict the API because it would be a breaking change.

Reply:
What do you think?

Reply:
That sound sreasonable yeah